operators/src/plot/temporal_vector_line_plot.rs

```diff
@@ -0,0 +1,369 @@
+        });
+
+        Ok(TypedPlotQueryProcessor::JsonVega(processor))
+    }
+}
+
+/// A query processor that calculates the `TemporalVectorLinePlot` on its input.
+pub struct FeatureAttributeValuesOverTimeQueryProcessor<G>
+where
+    G: Geometry + ArrowTyped + Sync + Send + 'static,
+{
+    params: FeatureAttributeValuesOverTimeParams,
+    features: Box<dyn VectorQueryProcessor<VectorType = FeatureCollection<G>>>,
+}
+
+#[async_trait]
+impl<G> PlotQueryProcessor for FeatureAttributeValuesOverTimeQueryProcessor<G>
+where
+    G: Geometry + ArrowTyped + Sync + Send + 'static,
+{
+    type OutputFormat = PlotData;
+
+    fn plot_type(&self) -> &'static str {
+        FEATURE_ATTRIBUTE_OVER_TIME_NAME
+    }
+
+    async fn plot_query<'a>(
+        &'a self,
+        query: QueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> Result<Self::OutputFormat> {
+        let values = FeatureAttributeValues::<MAX_FEATURES>::default();
+
+        let values = self
+            .features
+            .vector_query(query, ctx)?
+            .fold(Ok(values), |acc, features| async {
+                match (acc, features) {
+                    (Ok(mut acc), Ok(features)) => {
+                        let ids = features.data(&self.params.id_column)?;
+                        let values = features.data(&self.params.value_column)?;
+
+                        for i in 0..features.len() {
+                            let id: String = match ids.get_unchecked(i) {
+                                FeatureDataValue::Int(v) => v.to_string(),
+                                FeatureDataValue::NullableInt(v) => v
+                                    .map(|v| v.to_string())
+                                    .ok_or(Error::FeatureDataValueMustNotBeNull)?,
+                                FeatureDataValue::Float(v) => v.to_string(),
+                                FeatureDataValue::NullableFloat(v) => v
+                                    .map(|v| v.to_string())
+                                    .ok_or(Error::FeatureDataValueMustNotBeNull)?,
+                                FeatureDataValue::Text(v) => v.clone(),
+                                FeatureDataValue::NullableText(v) => {
+                                    v.ok_or(Error::FeatureDataValueMustNotBeNull)?
+                                }
+                                FeatureDataValue::Category(v) => v.to_string(),
+                                FeatureDataValue::NullableCategory(v) => {
+                                    v.map(|v| v.to_string())
+                                        .ok_or(Error::FeatureDataValueMustNotBeNull)?
+                                }
+                            };
+
+                            let value: f64 = match values.get_unchecked(i) {
+                                FeatureDataValue::Int(v) => v as f64,
+                                FeatureDataValue::NullableInt(v) => v.map_or(0.0, |v| v as f64), // TODO: NAN better default?
```

Comment:
You are right, I guess we should leave all null-values out?

Could you create a test for a plot where null-values where left out and one for duplicate values?

Reply:
I handle nulls now: 92801e562d20b900275516583b8b8ab4b228334b
not sure about duplicate values. Should I check the whole vector if the exact timestamp with the same value already exists? Should I keep the vector sorted then?

Reply:
I guess you don't have to do anything if vega handles it somehow. That is why you could create a test and then we see the outcome. My guess is that it's fine for now.

Reply:
I only test the generated vega code (which of course will contain duplicates) and not the resulting diagram. How do you suggest to do that?

Reply:
I just paste the resulting definition to https://vega.github.io/editor/ and see if this works out.
If so, having the json definition as our test result is the same as saying "checked values with QGis".

Reply:
ok, I did that and the plots looks the same 7bf399c3fbc05ba4b2f755ea38a9004a4fb57280